csrc/scheduler/transpose.cpp

```diff
@@ -84,10 +84,26 @@ class DomainMap : public pointwise_utils::DomainMap {
     if (ref1 == nullptr || ref2 == nullptr) {
       return false;
     }
+    auto innermost2 = scheduler_utils::innerMostRootDim(ref2);
+    auto innermost1 = domain_map.getMappedRootDimIn(ref1, innermost2);
     // reference 1 is the global reference, so it must have dim mapped the
     // innermost dim of both groups
-    auto innermost2 = scheduler_utils::innerMostRootDim(ref2);
-    return domain_map.getMappedRootDimIn(ref1, innermost2) != nullptr;
+    if (innermost1 == nullptr) {
+      return false;
+    }
+    for (auto ref : grouped_inputs_outputs[0]) {
+      auto innermost = domain_map.getMappedRootDimIn(ref, innermost1);
+      if (innermost && innermost->isRFactorProduct()) {
```

Comment:
Why does this need to be an rfactor product?

Reply:
I think this is the case where I reject the fusion for transpose scheduler.

IIUC, this is where we have the mapped innermost dimension into a "merged" dimension? hence RFactorProduct, where transpose scheduler would try to split it and that's failing the propagation.
This was caught with test `FusionReshapeMagicSchedule10_CUDA` and this happens before Xiang explained to me what the limitation of our transform propagation has.

Hopefully I'm doing the right thing here.

Reply:
Actually since we now check for propagation across view op in the dry-run propagation. I'm guessing we don't need this any more. Let me remove it and see if the whole world would crash and burn.

Reply:
Oops, reverting this break test. `NVFuserTest.FusionReshapePersistentShmoo_CUDA`. I'm taking a look there.